src/index.js

```diff
@@ -19,7 +19,30 @@ class RpcEngine extends SafeEventEmitter {
     this._middleware.push(middleware)
   }
 
-  handle (req, cb) {
+  handle (req, callback) {
+    if (callback) {
+      return this.handleCallback(req, callback);
+    }
+    return this.handlePromise(req)
+  }
+
+  handlePromise (req) {
+    return new Promise ((resolve, reject) => {
+      const callback = (err, result) => {
+        if (err) return reject(err)
+        return resolve(result)
+      }
+
+      // batch request support
+      if (Array.isArray(req)) {
+        this._handleBatch(req, callback)
+      } else {
+        this._handle(req, callback)
+      }
```

Comment:
This should probably just call `handleCallback`.

Reply:
yeah use pify on the handleCallback
```js
return pify(cb => this.handleCallback(req, cb)()
```